chips/cc26x2/src/uart.rs

```diff
@@ -160,39 +185,83 @@ impl UART {
     }
 
     fn enable_interrupts(&self) {
-        // Disable all UART interrupts
-        self.registers.imsc.modify(Interrupts::ALL_INTERRUPTS::SET);
+        // clear all
+        self.registers.icr.write(Interrupts::ALL_INTERRUPTS::Clear);
+        // set only interrupts used
+        self.registers.imsc.modify(
+            Interrupts::RX::SET
+                + Interrupts::RX_TIMEOUT::SET
+                + Interrupts::END_OF_TRANSMISSION::SET,
+        );
     }
 
     /// Clears all interrupts related to UART.
     pub fn handle_interrupt(&self) {
```

Comment:
We're neglecting to explicitly check which interrupts have fired. Is that intentional because it's not worth the trouble of avoiding certain branches if could know, e.g., that the RX buffer is full?

Reply:
Good question. I guess I've picked this up from lots of vendor code in the past and always reasoned that if any of those flags have flipped, it should be ignored (as this is usually just called during initialization). 

But, now that you mention it, I'm putting some hidden behavior in this "enable_interrupts"... plus if my goal is to put my device in a known state, I should be purging RX buffers like in your example.

At the very least, I think I should perhaps rename the function to enable_interrupt_and_clear_state and perhaps purge RX buffers. I'm trying to think if there's anything else that might "carry state".